localization/ekf_localizer/src/ekf_localizer.cpp

```diff
@@ -111,13 +112,57 @@ EKFLocalizer::EKFLocalizer(const std::string & node_name, const rclcpp::NodeOpti
   pub_measured_pose_ = create_publisher<geometry_msgs::msg::PoseStamped>("debug/measured_pose", 1);
 }
 
+/*
+ * calcTimerRate
+ */
+double EKFLocalizer::calcTimerRate(size_t window_size) const
+{
+  if (time_buffer_.size() < window_size) {
+    return ekf_rate_;
+  }
+
+  const auto time_diff = (time_buffer_.back() - time_buffer_.front()).seconds();
+  const auto num_intervals = time_buffer_.size() - 1;
+
+  return static_cast<double>(num_intervals) / time_diff;
+}
+
+/*
+ * updatePredictFrequency
+ */
+void EKFLocalizer::updatePredictFrequency()
+{
+  size_t window_size = 10;
+  time_buffer_.push_back(clock_->now());
+  while (static_cast<int>(time_buffer_.size()) > window_size) {
+    time_buffer_.pop_front();
+  }
+
+  // Calc timer rate
+  double timer_rate = calcTimerRate(window_size);
```

Comment:
@kosuke55 Any reason for this timer smoothing?

Reply:
@TakaHoribe 
I just imitated `ros2 topic hz` or `topic_state_monitor`.
It's better to use the diff from the previous callback time and update every loop?

Reply:
In the `topic_tool` or `state_monitor`, the timer window is for a noise reduction or conversion in an easy-to-see format for users. In the EKF, it can handle any noisy frequency, and this smoothing will probably cause performance degradation.

Reply:
OK, I fixed it in https://github.com/autowarefoundation/autoware.universe/pull/399/commits/cd797c208070371272d9df54f5e9b047728d9115